File: 프로그래머스/unrated/181872. 특정 문자열로 끝나는 가장 긴 부분 문자열 찾기/특정 문자열로 끝나는 가장 긴 부분 문자열 찾기.java
class Solution {
    public String solution(String myString, String pat) {
      String answer = "";
        int index = myString.indexOf(pat);
        
        while (index != -1) {
            answer += myString.substring(0, index + pat.length());
            myString = myString.substring(index + pat.length());
            index = myString.indexOf(pat);
        }
        
        return answer;
    }
}

File: 프로그래머스/0/181872. 특정 문자열로 끝나는 가장 긴 부분 문자열 찾기/특정 문자열로 끝나는 가장 긴 부분 문자열 찾기.js
function solution(myString, pat) {
    let answer = "";
    let index = myString.indexOf(pat);

    while (index !== -1) {
        answer += myString.substring(0, index + pat.length);
        myString = myString.substring(index + pat.length);
        index = myString.indexOf(pat);
    }

    return answer;
}